src/repl/scala/tools/nsc/interpreter/ReplReporter.scala

```diff
@@ -63,7 +89,7 @@ class  ReplReporter(intp: IMain) extends ConsoleReporter(intp.settings, Console.
     else Console.println("[init] " + msg)
   }
 
-  override def displayPrompt() {
+  override def displayPrompt(): Unit = {
     if (intp.totalSilence) ()
     else super.displayPrompt()
   }
```

Comment:
From the reverted, `override def displayPrompt() = if (!intp.totalSilence) super.displayPrompt()`  There must be a word for this style of supplying explicit unit values. `Unit saturation`? There is a ticket where the lub crashes...